src/components/CodeOverview/index.tsx

```diff
@@ -0,0 +1,217 @@
+import * as React from 'react';
+import { withRouter, Link, RouteComponentProps } from 'react-router-dom';
+import chunk from 'lodash.chunk';
+import debounce from 'lodash.debounce';
+
+import { getCodeLineAnchor, getLines } from '../CodeView/utils';
+import styles from './styles.module.scss';
+import { gettext } from '../../utils';
+import { Version } from '../../reducers/versions';
+import LinterProvider, { LinterProviderInfo } from '../LinterProvider';
+import CodeLineShapes from '../CodeLineShapes';
+import {
+  AllLineShapes,
+  LineShapes,
+  generateLineShapes,
+} from '../CodeLineShapes/utils';
+
+export type PublicProps = {
+  content: string;
+  version: Version;
+};
+
+export type DefaultProps = {
+  _debounce: typeof debounce;
+  _window: {
+    addEventListener: typeof window.addEventListener;
+    removeEventListener: typeof window.removeEventListener;
+  };
+  overviewPadding: number;
+  rowTopPadding: number;
+  rowHeight: number;
+};
+
+export type Props = PublicProps & DefaultProps & RouteComponentProps;
+
+type State = {
+  // This is the height of the overview div in pixels.
+  overviewHeight: number | null;
+};
+
+export class CodeOverviewBase extends React.Component<Props, State> {
+  static defaultProps = {
+    _debounce: debounce,
+    _window: window,
+    // This is the padding of the overview container.
+    overviewPadding: 10,
+    rowTopPadding: 2,
+    // This is the height of the row, including rowTopPadding.
+    rowHeight: 10,
+  };
+
+  public state = { overviewHeight: null };
+
+  private overviewRef = React.createRef<HTMLDivElement>();
+
+  componentDidMount() {
+    const { _window } = this.props;
+    this.setOverviewHeight();
+
+    // When the user begins resizing, first clear the overview contents.
+    _window.addEventListener('resize', this.resetOverviewHeight);
+    // After a short delay, get the new height and re-render.
+    _window.addEventListener('resize', this.waitAndSetNewOverviewHeight);
+  }
+
+  componentWillUnmount() {
+    const { _window } = this.props;
+    _window.removeEventListener('resize', this.resetOverviewHeight);
+    _window.removeEventListener('resize', this.waitAndSetNewOverviewHeight);
+  }
+
+  resetOverviewHeight = () => {
+    this.setState({ overviewHeight: null });
+  };
+
+  setOverviewHeight = (ref = this.overviewRef) => {
+    if (ref && ref.current) {
+      this.setState({ overviewHeight: ref.current.clientHeight });
+    }
+  };
+
+  waitAndSetNewOverviewHeight = this.props._debounce(
```

Comment:
This doesn't seem to be covered by a test. Is there something we can do to test this?

Reply:
Ah, yes there is. I couldn't figure it out at first but I got it to work.